ARCHITECTURE.md

```diff
@@ -1,292 +1,118 @@
-
-Interfaces:
-
--   READS from `Storage` while crawling, to check for differences (via the `diffing service`?)
--   WRITES to `Storage` when new versions are found
-
-<a id="org5cbd32e"></a>
-
-### Storage
-
-HTML and other resources are stored in Google cloud storage, along with an accompanying database that tracks changes. Database schemas are not currently known.
-
-Interfaces:
-
--   Communicates with `Page Collection` as above
--   Provides responses to `Diffing Service` queries about individual pages (see below)
--   Provides lists of pages and versions to `Researcher UI` (see below)
-
-<a id="orgba1a9c9"></a>
-
-### Diffing Service
-
-This service accepts queries about individual pages (and perhaps large sets of pages) and returns diffs in JSON format. We can use PageFreezer&rsquo;s existing service or create a replacement, depending on how closely our needs are matched by PageFreezer&rsquo;s. [We already have a candidate](https://github.com/edgi-govdata-archiving/differ) (developed to overcome some shortcomings of PageFreezer&rsquo;s *external* diff service), but we should probably evaluate the more robust internal diff UI with which we will soon be provided.
-
-Interfaces:
-
--   Queries `Storage` DB (via SQL?)
--   Provides JSON object to `UI`
-
-<a id="org0ee60b9"></a>
-
-### Filtration Service
-
-This service uses a set of heuristics to filter changes out of the result set presented in the UI. The heuristics will include generic and domain-specific rules. Eventually we would like to replace the ad-hoc rules in this service with machine-learning algorithms.
-
-Interfaces:
-
--   Either queries Storage for new pages or receives notification from Page Collection  when new pages are added to Storage.
--   Assigns a priortiy or other labels that can be used to filter important/unimportant changes and, ideally, prioritize them by estimated importance.
-
-<a id="org3b5f7b7"></a>
-
-### User Interface
-
-This web-based interface presents end users (analysts) with views of page collections (e.g., all pages from a given domain that pass the `filtration Service` tests, above) and individual pages. Analysts will be able to tag particular changes as belonging to a variety of categories, to be used later in EDGI&rsquo;s weekly reports. We&rsquo;ll place a high premium on ease of use, and speed of movement through changed pages; we&rsquo;ll probably want to be at least somewhat clever in our use of prefetching, etc.
-
-Interfaces:
-
--   Requests individual page diffs from `diff service`
--   Requests full lists of changed pages from `Storage`
--   Writes results of analysts&rsquo; inspection to `monitoring-database`
-
-<a id="org1c8db9d"></a>
+<a id="0"></a>
+## Definition of Terms
 
-### Monitoring Database
+* Page: a web page crawled over time by one or more services like the Internet
```

Comment:
Even though it’s called “page” in our model, we should probably clarify here that it’s not just web pages — it’s any resource on the web. (HTML pages, PDFs, Excel spreadsheets, CSVs, images, etc.)

Reply:
Could we call it a "Snapshot" instead of "Page"? "Snapshot" isn't defined or used anywhere formally, and I think it is broad enough to encompass those other types of resources.

Reply:
I don’t think that would be a good term for this. A snapshot is pretty analogous to a version, but definitely not to a page, which is ultimately a URL. “Snapshot,” especially in an archiving context, would imply that it’s about what was at a URL _at a particular point in time_, and when we say “page” we very intentionally mean something _not_ bounded by time.

“Resource” would be the correct technical term here (it’s probably what we *should* have named this model if we had been thinking more broadly to begin with). Otherwise, maybe URL? (The quibble with URL was always that we wanted this to represent the same page over time even if it’s URL changed, but we don’t actually support that [or at least not yet]).

Reply:
Oh, yeah, I wasn’t at all suggesting you rename it. Only that you change the “web page” in the *description,* not the term you were describing.